libnmstate/ifaces/ifaces.py

```diff
@@ -380,9 +380,15 @@ def _mark_orphan_as_absent(self):
         for iface in self._kernel_ifaces.values():
             if iface.need_parent and (iface.is_desired or iface.is_changed):
                 parent_iface = self._get_parent_iface(iface)
-                if parent_iface is None or parent_iface.is_absent:
+                if (parent_iface and parent_iface.is_absent) or (
+                    parent_iface is None and not iface.is_desired
+                ):
                     iface.mark_as_changed()
                     iface.state = InterfaceState.ABSENT
+                elif parent_iface is None and iface.is_desired:
+                    raise NmstateValueError(
+                        f"Failed to find parent interface for {iface.name}"
+                    )
```

Comment:
The integration test added in this commit is covering this case. I don't know why codecov complains about this line :-(